source/tools/glkernel-cmd/KernelGeneration.cpp

```diff
@@ -0,0 +1,379 @@
+        }
+
+        // Sampling
+        else if (command == "sample.poisson_square")
+        {
+            processPoissonSquare(kernel, parseArg("num_probes", 32));
+        }
+        else if (command == "sample.stratified")
+        {
+            processStratified(kernel);
+        }
+        else if (command == "sample.hammersley")
+        {
+            processHammersley(kernel);
+        }
+        else if (command == "sample.hammersley_sphere")
+        {
+            processHammersleySphere(kernel);
+        }
+        else if (command == "sample.halton")
+        {
+            processHalton(kernel, parseArg("base1", 2), parseArg("base2", 2));
+        }
+        else if (command == "sample.halton_sphere")
+        {
+            processHaltonSphere(kernel, parseArg("base1", 2), parseArg("base2", 2));
+        }
+        else if (command == "sample.best_candidate")
+        {
+            processBestCandidate(kernel, parseArg("num_candidates", 32));
+        }
+        else if (command == "sample.n_rooks")
+        {
+            processNRooks(kernel);
+        }
+        else if (command == "sample.multi_jittered")
+        {
+            processMultiJittered(kernel);
+        }
+        else if (command == "sample.golden_point_set")
+        {
+            processGoldenPointSet(kernel);
+        }
+
+        // Scale
+        else if (command == "scale.range")
+        {
+            glkernel::scale::range(kernel, parseArg("range_to_lower", 0.0f), parseArg("range_to_upper", 1.0f), parseArg("range_from_lower", 0.0f), parseArg("range_from_upper", 1.0f));
+        }
+
+        // Sequence
+        else if (command == "sequence.uniform")
+        {
+            glkernel::sequence::uniform(kernel, parseArg("range_min", 0.0f), parseArg("range_max", 1.0f));
+        }
+
+        // Shuffle
+        else if (command == "shuffle.bucket_permutate")
+        {
+            glkernel::shuffle::bucket_permutate(kernel, parseArg("subkernel_width", 1), parseArg("subkernel_height", 1), parseArg("subkernel_depth", 1));
+        }
+        else if (command == "shuffle.bayer")
+        {
+            glkernel::shuffle::bayer(kernel);
+        }
+        else if (command == "shuffle.random")
+        {
+            glkernel::shuffle::random(kernel, parseArg("start", 1));
+        }
+
+        // Sort
+        else if (command == "sort.distance")
+        {
+            // TODO: read origin, needs to be based on T
+            glkernel::sort::distance(kernel, T());
+        }
+        else
+        {
+            std::cout << "Command " << command << " is unknown. It is skipped." << std::endl;
+        }
+    }
+
+    template <typename T>
+    void processCommands(glkernel::tkernel<T> & kernel, const cppexpose::VariantArray & commandArray)
+    {
+        for (auto& command : commandArray)
+        {
+            auto commandMap = command.asMap();
+
+            auto& element = *commandMap->begin();
```

Comment:
If the map is empty, the defererence operation is malformed on most compilers. You should check against the end pointer first.